docs/src/developers_guide/contributing_tests.rst

```diff
@@ -0,0 +1,262 @@
+
+
+New fixtures should always be considered for ``conftest`` when added. If it is
+decided that they are not suitably reusable, they can be placed within the
+local test file.
+
+`Parameterisation <https://docs.pytest.org/en/stable/example/parametrize.html>`_
+--------------------------------------------------------------------------------
+
+Though it is a useful tool, we should not be complicating tests to work around
+parameters; they should only be used when it is simple and apparent to implement.
+
+Where you are parameterising multiple tests with the same parameters, it is
+usually prudent to use the `parameterisation within fixtures
+<https://docs.pytest.org/en/stable/how-to/fixtures.html#parametrizing-fixtures>`_.
+When doing this, ensure within the tests that it is apparent that they are being
+parameterised, either within the fixture name or with comments.
+
+All parameterisation benefits from
+`ids <https://docs.pytest.org/en/stable/example/parametrize.html#different-options-for-test-ids>`_,
+and so should be used where possible.
+
+`Mocks <https://docs.pytest.org/en/stable/how-to/monkeypatch.html>`_
+--------------------------------------------------------------------
+
+Any mocking should be done with ``pytest.mock``, and monkeypatching where suitable.
+
+.. note::
+    If you think we're missing anything important here, please consider creating an
+    issue or discussion and share your ideas with the team!
+
+`Classes <https://docs.pytest.org/en/stable/getting-started.html#group-multiple-tests-in-a-class>`_
+---------------------------------------------------------------------------------------------------
+
+How and when to group tests within classes can be based on personal opinion,
+we do not deem consistency on this a vital concern.
+
+Naming Test Classes and Functions
+---------------------------------
+
+Within a test module each tested method may have corresponding test classes,
+for example:
+
+* ``Test_<name of public method>``
+* ``Test_<name of public method>__<aspect of method>``
+
+Within these test classes, the test methods must be named according
+to the aspect of the tested method which they address.
+
+**Examples**:
+
+All unit tests for :py:class:`iris.cube.Cube` reside in:
+
+    :literal:`lib/iris/tests/unit/cube/test_Cube.py`
+
+Within that file the tests might look something like:
+
+.. code-block:: python
+
+    # A single test for the Cube.xml() method.
+    def test_xml_some_general_stuff(self):
+        ...
+
+
+    # A single test for the Cube.xml() method, focussing on the behaviour of
+    # the checksums.
+    def test_xml_checksum_ignores_masked_values(self):
+        ...
+
+
+    # Tests for the Cube.add_dim_coord() method.
+    class Test_add_dim_coord:
+        def test_normal_usage(self):
+            ...
+
+        def test_coord_already_present(self):
+            ...
+
+Within this test module there may be test classes, for example:
```

Comment:
We're missing some context since we moved things around. This section is talking about testing of **functions**; we should make that clear with a subheading or a sentence.